### PR TITLE
Metadata mega-rename

### PR DESCRIPTION
**Pull request by smackesey**
Repo-wide rename of metadata and friends:

- event_metadata.py > metadata.py
- EventMetadataEntry > MetadataEntry
- XMetadataEntryData > XMetadataValue
- parse_metadata > normalize_metadata

## Test Plan

The existing test suite.

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6601** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6601" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Review by alangenfeld — CHANGES_REQUESTED** [review]
I expect there to be back compat test failures given these changes to serialized class artifacts. 

I think effectively we have two choices 
* ensure the new class names serialize identical to their previous names. This will allow events written in `>0.14` to be deserialized by `<0.14`
* add fallback handlers for the old names to remap them to the new ones. This is bare minimum as we must be able to read the old artifacts. If we only do this, `dagit`, `daemon` etc must be `>0.14` if any user code is `>0.14`

An example for serialize as old is here: https://github.com/dagster-io/dagster/pull/5467

**Review by alangenfeld — COMMENTED** [review]
I would advocate for the full back & forth compat to minimize potential thrash experienced by users.

File: docs/sphinx/sections/api/apidocs/ops.rst [hunk 1]
```diff
@@ -72,43 +72,54 @@ Event types
 Event metadata
```
Comment (sryza) [inline docs/sphinx/sections/api/apidocs/ops.rst:72]: We should rename this heading, right?

**Review by sryza — APPROVED** [review]
Bombs away!

**Pull request merged**